From 02c432d6e0cdf2d784942895b5bdef78c1d2ab7e Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <uwu@icenowy.me>
Date: Wed, 19 Jun 2024 10:57:36 +0800
Subject: [PATCH 0833/1206] riscv: dts: spacemit: fix PCIe lane number for deb1

K1-X deb1 board seems to have only one PCIe lane wired for PCIe C port,
as a mini PCIe slot.

Fix the PCIe lane number, otherwise the PCIe controller could not be
correctly brought up and no card will be detected.

Signed-off-by: Icenowy Zheng <uwu@icenowy.me>
Change-Id: I51f9ff456d9176f7828d20889603845df9ea4e0d
---
 arch/riscv/boot/dts/spacemit/k1-x_deb1.dts | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/riscv/boot/dts/spacemit/k1-x_deb1.dts b/arch/riscv/boot/dts/spacemit/k1-x_deb1.dts
index fdf9a3a52440..52b7dd6f1896 100644
--- a/arch/riscv/boot/dts/spacemit/k1-x_deb1.dts
+++ b/arch/riscv/boot/dts/spacemit/k1-x_deb1.dts
@@ -835,6 +835,7 @@ &pcie1_rc {
 &pcie2_rc {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_pcie2_4>;
+	num-lanes = <1>;
 	status = "okay";
 };
 
-- 
2.47.0

